rpcserver.go

```diff
@@ -473,6 +477,49 @@ func (r *rpcServer) listBalancesByFamilyKey(ctx context.Context,
 	return resp, nil
 }
 
+// ListUtxos lists the UTXOs managed by the target daemon, and the assets they
+// hold.
+func (r *rpcServer) ListUtxos(ctx context.Context,
+	_ *tarorpc.ListUtxosRequest) (*tarorpc.ListUtxosResponse, error) {
+
+	rpcAssets, err := r.fetchRpcAssets(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	managedUtxos, err := r.cfg.AssetStore.FetchManagedUTXOs(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	utxos := make(map[string]*tarorpc.ManagedUtxo)
+	for _, u := range managedUtxos {
+		utxos[u.OutPoint.String()] = &tarorpc.ManagedUtxo{
+			OutPoint:    u.OutPoint.String(),
+			AmtSat:      int64(u.OutputValue),
+			InternalKey: u.InternalKey.PubKey.SerializeCompressed(),
+			TaroRoot:    u.TaroRoot,
+		}
+	}
+
+	// Populate the assets managed by each UTXO.
+	for _, a := range rpcAssets {
+		op := a.ChainAnchor.AnchorOutpoint
+		utxo, ok := utxos[op]
+		if !ok {
+			return nil, fmt.Errorf("unable to find utxo %s for "+
+				"asset", op)
```

Comment:
nit: could also add the asset ID to the error message for more context.

Reply:
Done: https://github.com/lightninglabs/taro/pull/185/commits/36dd4b390ce24d480f8bdc9874c4875253277a1e